### PR TITLE
Fix query parameters avoiding duplicates

### PR DESCRIPTION
**Pull request by andre-bisa**
queryStringParameters and multiValueQueryStringParameters were merged together, but multiValueQueryStringParameters already contains all queryStringParameters

### CODE REVIEW

**Comment by armanbilge** [comment]
Woops, good catch! I think we have the same problem with headers, would you mind fixing that as well?

**Comment by andre-bisa** [comment]
@armanbilge I've fixed headers as well, considering only multivalues. I've also included tests for both (query params and headers)

File: lambda-http4s/src/main/scala/feral/lambda/http4s/ApiGatewayProxyHandler.scala [hunk 1]
```diff
@@ -114,6 +95,15 @@ object ApiGatewayProxyHandler {
...
     }
 
+  private def getMultiValueHeaders(
+      multiValueHeaders: Option[Map[CIString, List[String]]]): List[Header.Raw] = {
+    for {
+      headerMap <- multiValueHeaders.toList
+      (key, values) <- headerMap
+      headerRaw <- values.map(value => Header.Raw(key, value))
+    } yield headerRaw
+  }
```
Comment (armanbilge) [inline lambda-http4s/src/main/scala/feral/lambda/http4s/ApiGatewayProxyHandler.scala:105]: This implementation uses a lot of intermediate allocations, would you mind re-using the original `ListBuilder`-based implementation?

File: lambda-http4s/src/main/scala/feral/lambda/http4s/ApiGatewayProxyHandler.scala [hunk 2]
```diff
@@ -61,29 +62,15 @@ object ApiGatewayProxyHandler {
...
-        event.headers.foreach { h => h.foreachEntry(builder += Header.Raw(_, _)) }
-        event.multiValueHeaders.foreach { hMap =>
-          hMap.foreach {
-            case (key, values) =>
-              if (!event.headers.exists(_.contains(key))) {
-                values.foreach(value => builder += Header.Raw(key, value))
-              }
-          }
-        }
-        Headers(builder.result())
```
Comment (armanbilge) [inline lambda-http4s/src/main/scala/feral/lambda/http4s/ApiGatewayProxyHandler.scala:85]: This one.

Comment (andre-bisa) [reply lambda-http4s/src/main/scala/feral/lambda/http4s/ApiGatewayProxyHandler.scala]: Done

**Review by armanbilge — APPROVED** [review]
thanks for your work!

**Pull request merged**